### ISSUE TITLE
"continuation line over-indented" when comprehension is an argument

### ISSUE THREAD

**Issue by anntzer** [PyCQA/pycodestyle#680]
Consider
```
[obj for obj in iterator
 if some_long_cond()
    and some_other_cond()]
(obj for obj in iterator
 if some_long_cond()
    and some_other_cond())
sorted(obj for obj in iterator
       if some_long_cond()
          and some_other_cond())
```

I would consider that this is a reasonable, and perhaps even *the* correct way to indent such expressions, assuming you adhere to the "binary operators after linebreak" school and also assuming that the expressions are too long to fit in a single line.

Interestingly, pycodestyle is happy with the indenting of the first two expressions, but complains about the last one
```
test.py:9:11: E127 continuation line over-indented for visual indent
```
(Note that it is able to recognize the correct amount of indenting needed in the first two expressions -- adding or removing a space before "and" results in the same warning as for the third expression.)

**Comment by FichteFoll** [comment]
I think the fundamental construct here is the conditional. The proposed indentation doesn't work outside of comprehensions either:

    abc = (1 if a
                and 2
             else b)

I don't think PEP 8 mentions anything about this, but I haven't looked either.

**Comment by anntzer** [comment]
The conditional in
`(foo for foo in bar if quux)`
and in
`(foo if bar else quux)`
are quite different objects (one is a clause to a comprehension, the other a ternary).

**Comment by FichteFoll** [comment]
That is correct. I was just thinking that inline `if` wasn't considered for indentation rules at all.

**Comment by flutefreak7** [comment]
I'm getting a related problem with the following code:

```python
import pandas as pd
data = pd.DataFrame({'A': [1, 2, 3, 4, 5], 'B': [0, 0, 0, 0, 0], 'C': [4, 5, 6, 7, 8]})
nonzero = (data != 0)
print(nonzero)
```

I get E128 continuation line underindented on the print(col) line. Looks like the parenthesis around `data != 0` is confusing it somehow.

**Comment by asottile** [comment]
@flutefreak7 I cannot reproduce, please create a new issue with more information (version, etc.)
